chalk-solve/src/infer/unify.rs

```diff
@@ -523,36 +1001,56 @@ impl<'t, I: Interner> Unifier<'t, I> {
         Ok(())
     }
 
-    fn push_lifetime_eq_goals(&mut self, a: Lifetime<I>, b: Lifetime<I>) {
-        self.goals.push(InEnvironment::new(
-            self.environment,
-            WhereClause::LifetimeOutlives(LifetimeOutlives {
-                a: a.clone(),
-                b: b.clone(),
-            })
-            .cast(self.interner),
-        ));
-        self.goals.push(InEnvironment::new(
-            self.environment,
-            WhereClause::LifetimeOutlives(LifetimeOutlives { a: b, b: a }).cast(self.interner),
-        ));
+    fn push_lifetime_eq_goals(&mut self, variance: Variance, a: Lifetime<I>, b: Lifetime<I>) {
+        debug!(
+            "pushing lifetime eq goals for a={:?} b={:?} with variance {:?}",
+            a, b, variance
+        );
+        if matches!(variance, Variance::Invariant | Variance::Covariant) {
```

Comment:
```suggestion
        if matches!(variance, Variance::Invariant | Variance::Contravariant) {
```
`'a <: 'b` is taken to mean `'b: 'a` (the other `matches` also needs changing)